### ISSUE TITLE
New 1.0.0 update fails WASM/JS builds

### ISSUE THREAD

**Issue by vidikn** [InsertKoinIO/koin-compiler-plugin#40]
1.0.0-RC2 worked fine. On 1.0.0 this only happens for WasmJS builds.

```
java.lang.AssertionError: Different declarations with the same signatures were detected, but no diagnostics will be reported because none of those declarations have any source location info associated with them. This could happen because the declarations came from an external module, or were generated by a compiler plugin. The following declarations have conflicting signatures:
 - FUN name:injectedparams_io_liveselect_app_core_data_AdminRepository visibility:public modality:FINAL <> (supabaseClient:io.github.jan.supabase.SupabaseClient) returnType:kotlin.Unit
 - FUN name:injectedparams_io_liveselect_app_core_data_AdminRepository visibility:public modality:FINAL <> (supabaseClient:io.github.jan.supabase.SupabaseClient) returnType:kotlin.Unit

	at org.jetbrains.kotlin.backend.common.linkage.issues.SignatureClashDetector.reportSignatureClashTo(SignatureClashDetector.kt:71)
	at org.jetbrains.kotlin.backend.common.diagnostics.IdSignatureClashDetector.reportSignatureConflict(IdSignatureClashDetector.kt:20)
	at org.jetbrains.kotlin.backend.common.diagnostics.IdSignatureClashDetector.reportSignatureConflict(IdSignatureClashDetector.kt:14)
	at org.jetbrains.kotlin.backend.common.linkage.issues.SignatureClashDetector.reportErrorsTo(SignatureClashDetector.kt:53)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializedIrModule(IrModuleSerializer.kt:49)
	at org.jetbrains.kotlin.backend.common.serialization.SerializeModuleIntoKlibKt.serializeModuleIntoKlib(serializeModuleIntoKlib.kt:122)
	at org.jetbrains.kotlin.ir.backend.js.KlibKt.serializeModuleIntoKlib(klib.kt:512)
	at org.jetbrains.kotlin.cli.pipeline.web.WebKlibSerializationPipelinePhase.executePhase(WebKlibSerializationPipelinePhase.kt:42)
```

**Comment by Krupal32** [comment]
same type issue for iOS build too

**Comment by arnaudgiuliani** [comment]
Update for **1.0.1**:

**Partly fixed.** Two distinct problems were behind the WASM/JS failure:

1. **Duplicate `injectedparams_*` signatures** (the error in this report) — fixed in 1.0.1 (see #44). This was the wall most projects hit first.
2. **A second wall on wasmJs/JS**, exposed once the duplicate is gone:
   ```
   IllegalStateException: No file found for source null   (serializeModuleIntoKlib)
   ```

For #2, the plugin's own IR-generated hint files now carry a resolvable source, so **DSL-based projects compile to wasmJs** in 1.0.1. However, **annotation-based projects** (`@Module`/`@ComponentScan`) also generate cross-module discovery hints as **compiler-framework-created top-level declarations**, whose file has no settable source — and the **Kotlin 2.3.20** JS/Wasm KLIB serializer rejects that.

**This is a Kotlin compiler issue, not a plugin one:** [KT-82395](https://youtrack.jetbrains.com/issue/KT-82395). The Native backend tolerates it; only the 2.3.20 wasm/js serializer is strict. `FirFile.sourceFile` has no setter, so the plugin cannot supply the source.

**It is fixed in Kotlin 2.4.0.** Verified against 1.0.1:

| Kotlin | wasmJs (DSL) | wasmJs (annotations) |
|---|---|---|
| 2.3.20 | ✅ | ❌ KT-82395 |
| 2.4.0  | ✅ | ✅ |

**Recommendation:** for annotation-based wasmJs/JS, use **Kotlin 2.4.0** (supported by koin-compiler-plugin 1.0.1, which now works across Kotlin 2.3.20–2.4.x). Keeping this issue open to track the 2.3.20 case until KT-82395 is resolved upstream.
